components/infra-proxy-service/server/clients.go

```diff
@@ -2,31 +2,113 @@ package server
 
 import (
 	"context"
-	"sort"
+	"fmt"
 
 	chef "github.com/go-chef/chef"
 
 	"github.com/chef/automate/api/interservice/infra_proxy/request"
 	"github.com/chef/automate/api/interservice/infra_proxy/response"
 )
 
+// Client represents api client struct.
+type Client struct {
+	Name       string `json:"name"`
+	ClientName string `json:"clientname"`
+	OrgName    string `json:"orgname"`
+}
+
+// ClientResult list result from Search API
+type ClientResult struct {
+	Total int       `json:"total"`
+	Start int       `json:"start"`
+	Rows  []*Client `json:"rows"`
+}
+
 // GetClients get clients list
 func (s *Server) GetClients(ctx context.Context, req *request.Clients) (*response.Clients, error) {
 	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
 	if err != nil {
 		return nil, err
 	}
 
-	clients, err := c.client.Clients.List()
+	res, err := c.SearchClients(req.SearchQuery)
```

Comment:
can this be not like this : c.client.Clients.Search()

Reply:
making `c.client.Clients.Search()` is like adding the changes into `https://github.com/go-chef/chef/blob/master/client.go` but here we need to handle some cases based on our need, so added like this.

It would be some refactoring needed for making search API call for other objects like environments, roles, and data bag items.

Reply:
Although I don't believe in creating Tech Debut for future. But if still you want to only add code in this repo, then

You could create a separate file to extend features of ApiClientService and use that function which will still be `c.client.Clients.Search()`
Also, I could not see any unit tests.
Please let me know if I am mistaken.

Reply:
Sure will take a look how we can achieve it, thanks

Reply:
@vivek-yadav now, using `query, err := c.client.Search.NewQuery("INDEX", "SEARCH_TERM")` for searching please review it again, Thanks!